agent/reconciler/host_reconciler.go

```diff
@@ -311,6 +322,40 @@ func (r *HostReconciler) deleteEndpointIP(ctx context.Context, byoHost *infrastr
 	return nil
 }
 
+// getHostInfo gets the host platform details.
+func (r *HostReconciler) getHostInfo() (infrastructurev1beta1.HostInfo, error) {
+	hostInfo := infrastructurev1beta1.HostInfo{}
+
+	hostInfo.Architecture = runtime.GOARCH
+	hostInfo.OSName = runtime.GOOS
+
+	if distribution, err := getOperatingSystem(); err != nil {
+		return hostInfo, errors.Wrap(err, "failed to get host operating system image")
+	} else {
+		hostInfo.OSImage = distribution
+	}
+	return hostInfo, nil
+}
+
+// getOperatingSystem gets the name of the current operating system image.
+func getOperatingSystem() (string, error) {
```

Comment:
Are we re-inventing the wheel here? We have something similar that installer does in order to determine if pre-requisites are met. Check this -
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/6ba32a1927018c06008a5cd38606cc35337a0065/agent/installer/os_detector.go#L26-L28

Reply:
Its somewhat similar but this returns a normalised string of os_version_arch, but we want these separately as different fields. We can either introduce similar fields to this struct and then reuse it or would have to re-parse the string again to break it into the fields. First option seems better. WDYT?

Reply:
Made use of the existing os detector by adding a new function for this use case.